app/controllers/contacts_controller.rb

```diff
@@ -3,7 +3,9 @@ class ContactsController < ApplicationController
 
   # GET /contacts or /contacts.json
   def index
-    @contacts = Contact.all
+  @q = Contact.ransack(params[:q])
+  @contacts = @q.result.where(user_id: current_user).order(:last_name)
```

Comment:
Could you instead make use of your associations here

```suggestion
  @q = current_user.contacts.order(:last_name).ransack(params[:q])
  @contacts = @q.result
```